frontend/packages/core/src/NPS/feedback.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useState } from "react";
+import { clutch as IClutch } from "@clutch-sh/api";
+import styled from "@emotion/styled";
+import { Grid as MuiGrid } from "@material-ui/core";
+import MuiSuccessIcon from "@material-ui/icons/CheckCircle";
+import { debounce } from "lodash";
+import { v4 as uuid } from "uuid";
+
+import { userId } from "../AppLayout/user";
+import { Button } from "../button";
+import { Alert } from "../Feedback";
+import { TextField } from "../Input";
+import { client } from "../Network";
+import type { ClutchError } from "../Network/errors";
+import { Typography } from "../typography";
+
+import EmojiRatings, { Rating } from "./emojiRatings";
+
+/** Interfaces */
+
+interface FeedbackOptions {
+  origin: "WIZARD" | "ANYTIME";
+  onSubmit?: (submit: boolean) => void;
+}
+
+// Defaults in case of API failure
+export const defaults: IClutch.feedback.v1.ISurvey = {
+  prompt: "Rate Your Experience",
+  freeformPrompt: "What would you recommend to improve this?",
+  ratingLabels: [
+    {
+      emoji: 1,
+      label: "bad",
+    },
+    {
+      emoji: 2,
+      label: "ok",
+    },
+    {
+      emoji: 3,
+      label: "great",
+    },
+  ],
+};
+
+const StyledButton = styled(Button)({
+  fontSize: "14px",
+  padding: "0 8px",
+  height: "18px",
+});
+
+const StyledTextField = styled(TextField)({
+  marginTop: "15px",
+  ".MuiInputBase-root": {
+    fontSize: "14px",
+  },
+});
+
+/**
+ * NPS feedback component which is the base for both Wizard and Anytime
+ * Will fetch given survey options from the server based on the given origin
+ * Then display a feedback component based on the given emoji ratings
+ *
+ * @param opts Available feedback options
+ * @returns NPSFeedback component
+ */
+const NPSFeedback = (opts: FeedbackOptions) => {
+  const [hasSubmit, setSubmit] = useState<boolean>(false);
+  const [selected, setSelected] = useState<Rating>(null);
+  const [freeformFeedback, setFreeformFeedback] = useState<string>("");
+  const [error, setError] = useState<boolean>(false);
+  const [survey, setSurvey] = useState<IClutch.feedback.v1.ISurvey>({});
+  const [requestId, setRequestId] = useState<string>("");
+  const maxLength = 180;
+  const debounceTimer = 500;
+
+  const trimmed =
+    freeformFeedback.trim().length > maxLength
+      ? `${freeformFeedback.trim().substring(0, maxLength - 3)}...`
+      : freeformFeedback;
+
+  /** Property objects used to extend components and remove console warnings */
+  const AlertProps = {
+    iconMapping: {
+      info: <MuiSuccessIcon style={{ color: "#3548d4" }} />,
+    },
+    style: {
+      margin: "32px",
+      alignItems: "center",
+    },
+  };
```

Comment:
Just curious, is this because these components don't have support for these props?

Reply:
Pretty much. Throws a TS error:

```
error TS2322: Type '{ children: Element; severity: "info"; style: { margin: string; alignItems: string; }; }' is not assignable to type 'IntrinsicAttributes & AlertProps & { children?: ReactNode; }'.
  Property 'style' does not exist on type 'IntrinsicAttributes & AlertProps & { children?: ReactNode; }'.
```

Plan is to go and fix it but it started getting a bit out of scope for this one.